lib/modules/datasource/crate/index.ts

```diff
@@ -371,4 +377,39 @@ export class CrateDatasource extends Datasource {
 
     return [packageName.slice(0, 2), packageName.slice(2, 4), packageName];
   }
+
+  private static releaseTimestampSchema = z
+    .object({
+      version: z.object({
+        created_at: z.string(),
+      }),
+    })
+    .transform(({ version: { created_at } }) => created_at);
+
+  @cache({
+    namespace: `datasource-crate`,
+    key: (
+      { registryUrl, packageName }: PostprocessReleaseConfig,
+      { version }: Release,
+    ) => `postprocessRelease:${registryUrl}:${packageName}:${version}`,
+    ttlMinutes: 24 * 60,
```

Comment:
Are releases immutable on crates.io? If so then we should think about the maximum feasible cache time for this - perhaps a week?

Reply:
We should also make sure we've considered the scenario where we get a temporary error (including 429) and don't cache that for long

Reply:
It could be yanked, and also there is `updated_at` field, so let's not consider it immutable.

> We should also make sure we've considered the scenario where we get a temporary error (including 429) and don't cache that for long

With current cache decorator implementation, it's hard to set the TTL a posteriori. If the error was thrown, we could either not cache it, or we could catch and return the cacheable null (without distinguishing the reason).

Reply:
If updated_at is only triggered for fields we don't care about, then we can ignore updated_at. And wouldn't the `yanked` status be reflected in our regular crates.io lookup which happens before this API call?

Reply:
Okay, you're right